specs/tests/Spec.Blake2.Test.fst

```diff
@@ -276,108 +778,67 @@ let test6_expected : lbytes 64 =
+ if result2 then begin  IO.print_string "\nSuccess!\n"; true end
+ else begin IO.print_string "\nFailure :(\n";
 
-  IO.print_string "\n4. Expected: ";
-  List.iter (fun a -> IO.print_uint8 (u8_to_UInt8 a)) (to_list test4_expected);
-
-  //
-  // TEST 5
-  //
-  IO.print_string "\n\nTEST 5";
-
-  let test5_result : lbytes 64 =
-    Spec.Blake2.blake2b test5_plaintext 0 (of_list []) 64
-  in
-  let result5 = for_all2 (fun a b -> uint_to_nat #U8 a = uint_to_nat #U8 b) test5_expected test5_result in
-
-  IO.print_string "\n5. Result  : ";
-  List.iter (fun a -> IO.print_uint8 (u8_to_UInt8 a)) (to_list test5_result);
-
-  IO.print_string "\n5. Expected: ";
-  List.iter (fun a -> IO.print_uint8 (u8_to_UInt8 a)) (to_list test5_expected);
-
-
-  //
-  // TEST 6
-  //
-  IO.print_string "\n\nTEST 6";
-
-  let test6_result : lbytes 64 =
-    Spec.Blake2.blake2b test6_plaintext 64 test6_key 64
-  in
-  let result6 = for_all2 (fun a b -> uint_to_nat #U8 a = uint_to_nat #U8 b) test6_expected test6_result in
+  IO.print_string "\n2. Result  : ";
+  List.iter (fun a -> IO.print_uint8 (u8_to_UInt8 a)) (to_list result);
 
-  IO.print_string "\n6. Result  : ";
-  List.iter (fun a -> IO.print_uint8 (u8_to_UInt8 a)) (to_list test6_result);
+  IO.print_string "\n2. Expected: ";
+  List.iter (fun a -> IO.print_uint8 (u8_to_UInt8 a)) (to_list expected);
+ false end
 
-  IO.print_string "\n6. Expected: ";
-  List.iter (fun a -> IO.print_uint8 (u8_to_UInt8 a)) (to_list test6_expected);
 
+let test () =
+  let emp_key : lbytes 0 = (assert_norm (List.Tot.length ([] <: list uint8)<= max_size_t); of_list []) in
+  let result1 = blake2s_test 1 test1_plaintext emp_key test1_expected in
+  let result2 = blake2s_test 2 test2_plaintext test2_key test2_expected in
+  let result3 = blake2s_test 3 test3_plaintext test3_key test3_expected in
+  let result4 = blake2s_test 4 test4_plaintext test4_key test4_expected in
+  let result7 = blake2s_test 7 test7_plaintext test7_key test7_expected in
+  let result8 = blake2s_test 8 test8_plaintext test8_key test8_expected in
+  let result9 = blake2s_test 9 test9_plaintext test9_key test9_expected in
+  let result10 = blake2s_test 10 test10_plaintext test10_key test10_expected in
+  let result11 = blake2s_test 11 test11_plaintext test11_key test11_expected in
+
+  let result5 = blake2b_test 5 test5_plaintext emp_key test5_expected in
+  let result6 = blake2b_test 6 test6_plaintext test6_key test6_expected in
+  let result12 = blake2b_test 12 test12_plaintext test12_key test12_expected in
+  let result13 = blake2b_test 13 test13_plaintext test13_key test13_expected in
+  let result14 = blake2b_test 14 test14_plaintext test14_key test14_expected in
+  let result15 = blake2b_test 15 test15_plaintext test15_key test15_expected in
+  let result16 = blake2b_test 16 test16_plaintext test16_key test16_expected in
   //
   //
   // RESULT
   //
   if result1 && result2 && result3 && result4 && result5 && result6 then
```

Comment:
you are missing some test failure here

I would suggest chaining, e.g. 
```
let result = true in
let result = result && blake2s_test 1 ... in
let result = result && blake2s_test 2 ... in

if result then
  print "all tests successful"
else
 print "some test failed"
result

Reply:
note that you should be able to throw exceptions since test () is likely in the ML effect -- if that's a style you prefer, of course.

Reply:
I was trying not to short-circuit tests, since sometimes having failures in some tests but not others gives you more information. I will fix it so it captures all results.